resources/store/index.js

```diff
@@ -99,21 +138,26 @@ module.exports = new Vuex.Store( {
 		 * @param {string} tab
 		 */
 		setTab: function ( state, tab ) {
-			if ( state.tabs.indexOf( tab ) === -1 ) {
-				throw new Error( 'invalid tab' );
-			}
-
+			ensureTabExists( state, tab );
 			state.currentTab = tab;
 		},
 
 		/**
 		 * Sets the pending state
 		 *
 		 * @param {Object} state
-		 * @param {bool} pending
+		 * @param {Object} payload
+		 * @param {bool} payload.pending
+		 * @param {string} [payload.queue]
 		 */
-		setPending: function ( state, pending ) {
-			state.pending = !!pending;
+		setPending: function ( state, payload ) {
+			if ( payload.queue ) {
+				ensureTabExists( state, payload.queue );
+				state.pending[ payload.queue ] = !!payload.pending;
```

Comment:
I'm not quite following the use of `!!` here

Reply:
Good catch, I think this was left over from a previous toggle implementation where I wanted to coerce to boolean. Here it should probably just be:

```
state.pending[ payload.queue ] = true
```

Reply:
We were chatting about this in slack, but I thought I'd mention some of the points here in case anyone else is reading.

My philosophy with Vuex is to try to use mutations as gate-keepers for type sanity (maybe in the Glorious Typescript Future this won't be necessary any more). Since mutations are the only things allowed to actually touch the state of our store, you can ensure that the values there are **always** values of a given type by doing things like what I'm doing here.

Rather than allowing a (potentially arbitrary) property in our payload to get assigned as the value of `state.pending`, using `!!` ensures that we'll always have a boolean here no matter what.

Similarly, the mutations that add/remove images only rely on array methods like push and shift. Instead of writing an `addImages` method that takes an argument we hope is an array and commits it as the new value, the approach here is to push or shift each image one at a time, inside of a map or loop if necessary. Similar reasoning is behind the `ensureTabExists` helper function that throws errors the minute anything is amiss.

It's a little extra ceremony here, but the hope is that it will help to preserve our sanity down the road.